docs/proposals/configmap-rollout.md

```diff
@@ -0,0 +1,171 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# ConfigMap management
+
+## Abstract
+
+A proposal for implementing management of ConfigMaps in the context of a
+[Deployment](http://kubernetes.io/docs/user-guide/deployments/).
+
+Users should be able to have their apps automatically pick up changes to
+their configuration and also have the ability to rollback to previous
+configuration similar to the ability to rollback to previous pod template
+revisions.
+
+Currently, in order to apply updated configuration into a running app, users
+need to 1) either do it manually: copy the ConfigMap that holds their old
```

Comment:
We should lay out clearly the conditions that would make a controller set an ownerRef on a config map, since that has the potential to delete user-created data unintentionally.